source/common/network/socket_option_impl.cc

```diff
@@ -34,6 +36,26 @@ bool SocketOptionImpl::setOption(Socket& socket, Socket::SocketState state) cons
     }
   }
 
+  if (state == Socket::SocketState::Listening) {
```

Comment:
I think this should live in `ListenerSocketOption`, since it is specific to the listener side.

Reply:
Agreed

Reply:
ListenerSocketOption is addressed